File: backend/src/models/Account.js
const mongoose = require('mongoose');

const accountSchema = new mongoose.Schema({
  userId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  name: { type: String, required: true, trim: true },
  type: {
    type: String,
    enum: ['cash', 'bank', 'e-wallet', 'credit-card', 'savings'],
    default: 'cash',
  },
  balance: { type: Number, default: 0 },
  currency: { type: String, default: 'IDR' },
  color: { type: String, default: '#1890ff' },
  icon: { type: String, default: 'wallet' },
  isActive: { type: Boolean, default: true },
}, { timestamps: true });

module.exports = mongoose.model('Account', accountSchema);

File: mobile/lib/models/account.dart
class Account {
  final String id;
  final String name;
  final String type;
  final double balance;
  final String currency;
  final String color;
  final String icon;
  final bool isActive;

  Account({
    required this.id,
    required this.name,
    required this.type,
    required this.balance,
    required this.currency,
    required this.color,
    required this.icon,
    required this.isActive,
  });

  factory Account.fromJson(Map<String, dynamic> j) => Account(
        id: j['id'] ?? '',
        name: j['name'] ?? '',
        type: j['type'] ?? 'cash',
        balance: (j['balance'] as num?)?.toDouble() ?? 0,
        currency: j['currency'] ?? 'IDR',
        color: j['color'] ?? '#1890ff',
        icon: j['icon'] ?? 'wallet',
        isActive: j['isActive'] ?? true,
      );

  Map<String, dynamic> toJson() => {
        'name': name,
        'type': type,
        'balance': balance,
        'currency': currency,
        'color': color,
        'icon': icon,
      };
}
